src/badguy/icecrusher.cpp

```diff
@@ -297,13 +357,23 @@ IceCrusher::update(float dt_sec)
   switch (state) {
     case IDLE:
       m_col.set_movement(Vector (0, 0));
-      if (found_victim_down() && !sideways)
+      if (found_victim_up() && !sideways && m_flip != NO_FLIP)
+        set_state(CRUSHING_UP);
+      else if (found_victim_down() && !sideways && m_flip == NO_FLIP)
         set_state(CRUSHING);
-		  if (found_victim_right() && sideways)
+      else if (found_victim_right() && sideways)
```

Comment:
Careful - does this still allow players to walk atop a crusher? Many levels depend on that functionality, such as Ice in the Hole and a secret area in the Rooted Forest

Reply:
I think I checked that, but I will check again just to be sure.

Reply:
I checked it, it works fine.